src/components/KFileUpload/KFileUpload.vue

```diff
@@ -235,6 +250,33 @@ const resetInput = (): void => {
 
   emit('file-removed')
 }
+
+const setLabelAttributes = () => {
+  /**
+   * Temporary fix for the issue where we can't use v-bind-once to pass id to a custom element (KInput)
+   * TODO: remove this once useId is released in Vue 3.5
+   */
+  if (!attrs.id) {
+    const inputElementId = fileInputElement.value?.$el?.querySelector('input')?.id
+
+    if (inputElementId) {
+      labelElement.value?.$el?.setAttribute('for', inputElementId)
+    }
+  }
+}
+
+watch(fileInputKey, async () => {
+  await nextTick()
```

Comment:
Why does this code need to be in a separate watcher instead of in the one below that triggers the key to increment?